misc/custom_promise_types/cfengine.py

```diff
@@ -2,6 +2,7 @@
 import json
 import traceback
 
+_LOG_LEVELS = {level: idx for idx, level in enumerate(("critical", "error", "warning", "notice", "info", "verbose", "debug"))}
```

Comment:
Wow, this is some high level python magic 👏

Reply:
This is what people call Python these days. :wink: